pkg/expr/nodes.go

```diff
@@ -111,6 +121,29 @@ func buildCMDNode(rn *rawNode, toggles featuremgmt.FeatureToggles) (*CMDNode, er
 		CMDType: commandType,
 	}
 
+	if toggles.IsEnabledGlobally(featuremgmt.FlagExpressionParser) {
+		rn.QueryType, err = getExpressionCommandTypeString(rn.Query)
+		if err != nil {
+			return nil, err // should not happen because the command was parsed first thing
+		}
+
+		// NOTE: this structure of this is weird now, because it is targeting a structure
+		// where this is actually run in the root loop, however we want to verify the individual
+		// node parsing before changing the full tree parser
+		reader, err := NewExpressionQueryReader(toggles)
+		if err != nil {
+			return nil, err
+		}
+
+		iter := jsoniter.ParseBytes(jsoniter.ConfigDefault, rn.QueryRaw)
+		q, err := reader.ReadQuery(rn, jsonitersdk.NewIterator(iter))
```

Comment:
```suggestion
		iter, err := jsonitersdk.ParseBytes(jsoniter.ConfigDefault, rn.QueryRaw)
		if err != nil {
			return nil, err
		}
		q, err := reader.ReadQuery(rn, iter)
```

Then also remove jsoniter import (could use that name for sdk import if you like :shrug:)

Reply:
hymm that did not work -- jsonitersdk requires an instance 🤔